### PR TITLE
Feature/secretsmanager

### LINKED ISSUE #111: Upgrade secrets manager dependency
Currently when I generate a driven adapter which uses secrets, this dependency can not be downloaded from maven, I suggest to change this dependency `compile 'co.com.bancolombia:secretsmanager:2.1.0'` for the dependency described in the open source project [secrets-manager](https://github.com/bancolombia/secrets-manager).

### PR DESCRIPTION
**Pull request by andmagom**
## Description
New secrets adapter

## Fixes
* Fix #111 : Secrets Manager version upgraded 
* Fix: Mongo folder is not created when secrets  are created

## Category
- [x] Feature
- [x] Fix

## Checklist
- [x] The pull request is complete according to the [guide of contributing](https://github.com/bancolombia/scaffold-clean-architecture/wiki/Contributing)
- [x] Automated tests are written
- [x] The documentation is up-to-date
- [x] the version of the build.gradle, readme.md and gradle.properties was increased
- [x] The pull request has a descriptive title that describes what has changed, and provides enough context for the changelog

### CODE REVIEW

File: src/main/java/co/com/bancolombia/factory/adapters/DrivenAdapterSecrets.java [hunk 1]
```diff
@@ -4,19 +4,28 @@
...
     public void buildModule(ModuleBuilder builder) throws IOException, CleanException {
-        if (builder.getBooleanParam("include-secret")) {
+        builder.loadPackage();
+        Logger logger = builder.getProject().getLogger();
+        String secretLibrary = "";
+        if (builder.isReactive()) {
+            secretLibrary = "aws-secrets-manager-async";
+            builder.setupFromTemplate("driven-adapter/secrets-reactive");
+        } else {
+            secretLibrary = "aws-secrets-manager-sync";
```
Comment (santitigaga) [inline src/main/java/co/com/bancolombia/factory/adapters/DrivenAdapterSecrets.java:21]: would assign secretLibrary variable out of if scope

File: src/main/resources/driven-adapter/mongo-reactive/config/mongo-config.java.mustache [hunk 2]
```diff
@@ -17,9 +17,9 @@ public class MongoConfig {
 
     {{#include-secret}}
     @Bean
-    public MongoDBSecret dbSecret(@Value("${aws.secretName}") String secret, GenericManager manager)
+    public MongoDBSecret dbSecret(@Value("${aws.secretName}") String secret, GenericManagerAsync manager)
             throws SecretException {
-        return manager.getSecretModel(secret, MongoDBSecret.class);
+        return manager.getSecret(secret, MongoDBSecret.class).block();
```
Comment (santitigaga) [inline src/main/resources/driven-adapter/mongo-reactive/config/mongo-config.java.mustache:22]: call block function is required?

**Review by santitigaga — APPROVED** [review]

File: src/main/resources/driven-adapter/mongo-reactive/config/mongo-config.java.mustache (hunk 2 above)
Comment (andmagom) [reply src/main/resources/driven-adapter/mongo-reactive/config/mongo-config.java.mustache:22]: > call block function is required?
The Client mongo has to be changed first... Issue  #128

Comment (santitigaga) [reply src/main/resources/driven-adapter/mongo-reactive/config/mongo-config.java.mustache:22]: ok

**Pull request merged**